compiler/noirc_evaluator/src/ssa/ir/dfg.rs

```diff
@@ -619,17 +646,21 @@ impl DataFlowGraph {
 
     /// True if the given ValueId refers to a (recursively) constant value
     pub(crate) fn is_constant(&self, argument: ValueId) -> bool {
-        match &self[self.resolve(argument)] {
+        let argument = self.resolve(argument);
+        if self.is_global(argument) {
+            return true;
+        }
```

Comment:
This is a bit odd when `self.is_global` just matches on the value for a `Value::Global`, I think it'd be clearer to just have the Global match case below return true instead of unreachable

Reply:
I chose to separately check as I would need to special case `Value::Instruction`. I can switch to checking inside the match cases though.

Reply:
I have switched it